web/src/recent_topics_ui.js

```diff
@@ -728,6 +728,22 @@ function topic_sort(a, b) {
     return sort_comparator(topic_sort_key(a), topic_sort_key(b));
 }
 
+function unread_count(conversation_data) {
+    const message = message_store.get(conversation_data.last_msg_id);
+    return message_to_conversation_unread_count(message);
+}
+
+function unread_sort(a, b) {
+    const a_unread_count = unread_count(a);
+    const b_unread_count = unread_count(b);
+    if (a_unread_count < b_unread_count) {
+        return 1;
+    } else if (a_unread_count === b_unread_count) {
+        return a.last_msg_id < b.last_msg_id ? 1 : -1;
+    }
+    return -1;
+}
```

Comment:
I think we can tweak this function a little to make it more readable.

```suggestion
function unread_sort(a, b) {
    const a_unread_count = unread_count(a);
    const b_unread_count = unread_count(b);

    if (a_unread_count !== b_unread_count) {
        return b_unread_count - a_unread_count;
    }
    return b.last_msg_id - a.last_msg_id;
}
```

Reply:
Replaced, thanks